### PR TITLE
Add support for cc.com movies

### LINKED ISSUE #1926: cc.com
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.01**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Description

https://www.cc.com/topic/a-clusterfunke-christmas

TV episodes work fine, but the movie comes back Unsupported URL

### Verbose log

```shell
C:\Users\Kevin\Downloads\yt>ytdl.exe -Uv https://www.cc.com/movies/tkp406/a-clue
sterfuenke-christmas
[debug] Command-line config: ['-Uv', 'https://www.cc.com/movies/tkp406/a-clueste
rfuenke-christmas']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8 (No ANSI), err utf-8 (No A
NSI), pref cp1252
[debug] yt-dlp version 2021.12.01 [91f071a] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev (setts), ffprobe 4.4-fu
ll_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2021.12.01, Current version: 2021.12.01
yt-dlp is up to date (2021.12.01)
[debug] [generic] Extracting URL: https://www.cc.com/movies/tkp406/a-cluesterfue
nke-christmas
[generic] a-cluesterfuenke-christmas: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] a-cluesterfuenke-christmas: Downloading webpage
[generic] a-cluesterfuenke-christmas: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.cc.com/movies/tkp406/a-cluesterfuenke-christ
mas
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1329, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1398, in __extract_info
  File "yt_dlp\extractor\common.py", line 597, in extract
  File "yt_dlp\extractor\generic.py", line 3813, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.cc.com/movies/tkp406
/a-cluesterfuenke-christmas
```

### PR DESCRIPTION
**Pull request by sqrtNOT**
Extraction logic is identical to what we already have the url just didn't match.

Fixes #1926 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)

</details><details><summary>Copilot Summary</summary>  

### Summary
🎬🆕✅

Improved `ComedyCentralIE` extractor to handle movie URLs. Added a test case for `yt_dlp/extractor/comedycentral.py`.

> _Oh we are the coders of the Comedy Central_
> _We scrape the movies from the web so fine_
> _We update the `ComedyCentralIE` extractor_
> _And we add a test case on the count of three_

### Walkthrough
* Update the URL pattern to support movie URLs ([link](https://github.com/yt-dlp/yt-dlp/pull/7108/files?diff=unified&w=0#diff-c3cd878bca61955700d8d35d88132a1a76a6b8ea2b6a81a66c8dd80c289bae83L5-R5))
* Add a test case for movie extraction ([link](https://github.com/yt-dlp/yt-dlp/pull/7108/files?diff=unified&w=0#diff-c3cd878bca61955700d8d35d88132a1a76a6b8ea2b6a81a66c8dd80c289bae83R28-R30))

</details>

### CODE REVIEW

**Comment by sqrtNOT** [comment]
It compiles just fine on my machine not sure what's going on with the test suite.

**Comment by bashonly** [comment]
The test URL you added is not matched by the extractor

`https://www.cc.com/movies/tkp406/a-cluesterfuenke-christmas` is the example URL that should be supported, yeah?

**Comment by sqrtNOT** [comment]
> The test URL you added is not matched by the extractor
> 
> `https://www.cc.com/movies/tkp406/a-cluesterfuenke-christmas` is the example URL that should be supported, yeah?

It should match up to the "https://www.cc.com/movies/tkp406" at least. The extractor never cared about anything after the id as evidenced by the other test urls:
```
http://www.cc.com/video-clips/5ke9v2/the-daily-show-with-trevor-noah-doc-rivers-and-steve-ballmer---the-nba-player-strike
http://www.cc.com/episodes/pnzzci/drawn-together--american-idol--parody-clip-show-season-3-ep-314
https://www.cc.com/video/k3sdvm/the-daily-show-with-jon-stewart-exclusive-the-fourth-estate
```

I also downloaded the movie locally just now with the changes. Without them it falls back to the generic extractor and gives an unsupported url error.

File: yt_dlp/extractor/comedycentral.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ class ComedyCentralIE(MTVServicesInfoExtractor):
     }, {
         'url': 'https://www.cc.com/collection-playlist/cosnej/stand-up-specials/t6vtjb',
         'only_matching': True,
+    }, {
+        'url': 'https://www.cc.com/topic/a-clusterfunke-christmas',
```
Comment (bashonly) [inline yt_dlp/extractor/comedycentral.py:29]: so what I was saying is do this:
```suggestion
        'url': 'https://www.cc.com/movies/tkp406/a-cluesterfuenke-christmas',
```

Comment (sqrtNOT) [reply yt_dlp/extractor/comedycentral.py]: Oh rip I copied the wrong url. Sorry about that and good catch.

**Review by bashonly — APPROVED** [review]

**Pull request merged**